extensions/core/runtime/src/main/java/io/quarkiverse/cxf/vertx/http/client/VertxHttpClientHTTPConduit.java

```diff
@@ -657,11 +694,15 @@ void finishRequest(HttpClientRequest req, Buffer buffer) {
 
         }
 
+        private static int performedRetransmits(List<URI> retransmits) {
+            /* The first element in the retransmits list is the original URI that we do not count as a retransmit */
+            return retransmits.size() - 1;
+        }
+
         void redirectRetransmit(URI newURL) throws IOException {
             if (Log.isDebugEnabled()) {
```

Comment:
The if statement could indeed be omitted, if the log statement params would all be some simple values. But in this case, there is `redirects.stream().map(URI::toString).collect(Collectors.joining(" -> "))` which costs some CPU cycles, but is useless in case the debug level is not active.

Reply:
I never thought about it from a performance POV, but that makes a lot of sense, thanks!